### ISSUE TITLE
Native Module cannot be null

### ISSUE THREAD

**Issue by GuiMend** [zo0r/react-native-push-notification#541]
Hi, I'm having some problems when I try to use **react-native run-ios**

react-native-cli: 2.0.1
react-native: 0.31.0
node: 7.6.0

I already have done:
- PushNotification's lib linked
- added in the Header Search
- Signed in with a valid team in "General->Signing"
- tried running with and without "Push Notifications" on in capabilities 
but I'm still getting this error.

One thing that I find quite strange, is that when I open Xcode and run the app it works, but if I try to use the cmd **react-native run-ios** it doesn't work. 

I would like to use the cmd **react-native run-ios** so I can debug my app in the simulator. Thanks

<img width="372" alt="screen shot 2017-08-31 at 9 30 06 am" src="https://user-images.githubusercontent.com/9381325/29923306-7495c610-8e2f-11e7-9313-0c77e6e82ca8.png">
<img width="721" alt="screen shot 2017-08-31 at 9 32 44 am" src="https://user-images.githubusercontent.com/9381325/29923305-7494120c-8e2f-11e7-9cc6-82ed5050c6ed.png">
<img width="609" alt="screen shot 2017-08-31 at 9 33 19 am" src="https://user-images.githubusercontent.com/9381325/29923307-7496718c-8e2f-11e7-998c-e2f68ce2f5d7.png">

I've already seen issues #160 and #279 and still have this problem =(

**Comment by JeffGuKang** [comment]
Same

**Comment by majd-kho** [comment]
@GuiMend im fairly sure that you cannot test push notification using a simulator. I ran in to the issue myself and I had to use a physical device. 
[stack exchange](https://apple.stackexchange.com/questions/42654/is-it-possible-to-send-push-notifications-to-the-iphone-simulator)

**Comment by GuiMend** [comment]
I found a way to solve my problem.... I was trying to enter the **debug mode** that I was only able to access through 'react-native run-ios' (that **cmd + d**), but I found how to enable this mode in the Xcode "run".

Next to the RUN / STOP button the is a button with your app's name.
1) Click your app's name
2) enter "Edit Scheme" (bottom of the picker)
3) select the "Run" tab
4) make sure "Build Configuration" is in "Debug"

All done, next time you run your app through Xcode "RUN" you should be able to enter the debug by clicking "cmd + d"

**Comment by Reiji87** [comment]
It is still happening on my side, i tried all the ways i can find... anyone has a solution/fix for this?

**Comment by wincod75** [comment]
If anyone else comes across this issue and seemingly nothing works, try running npm save "package" and "react-native link X5" at the "same time" vs saving then running link. No idea if this will actually help, but I tried all sorts of things and nothing worked. I manually installed, uninstalled, did a fresh npm install, etc etc etc.... nothing. Somehow I did this and magically the build was successful.
